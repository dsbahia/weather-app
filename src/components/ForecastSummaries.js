import React from "react";
import ForecastSummary from "./ForecastSummary";

function ForecastSummaries({ forecasts, onForecastSelect }) {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          description={forecast.description}
          icon={forecast.icon}
          temperature={forecast.temperature}
          onSelect={onForecastSelect}
        />
      ))}
    </div>
  );
}

export default ForecastSummaries;
