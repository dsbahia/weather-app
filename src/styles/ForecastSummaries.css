.forecast-summaries {
    display: flex;
    justify-content: space-between;
}

.button {
    background-color: #007bff;
    color: #fff;
    border: none;
    padding: 10px 20px;
    font-size: 16px;
    border-radius: 5px;
    cursor: pointer;
    transition: background-color 0.3s ease;
  }
  
  .button:hover {
    background-color: #0056b3;
  }