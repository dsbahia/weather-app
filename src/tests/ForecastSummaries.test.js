import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        description: "Stub description",
        icon: 800,
        temperature: {
          min: 12,
          max: 22,
        },
      },
      {
        date: 2222222,
        description: "Another stub description",
        icon: 700,
        temperature: {
          min: 15,
          max: 25,
        },
      },
    ],
    onForecastSelect: jest.fn(),
  };

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps.forecasts} />,
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
