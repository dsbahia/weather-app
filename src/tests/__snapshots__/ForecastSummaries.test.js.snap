// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Stub description"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Another stub description"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        25°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Another stub description
      </div>
    </div>
  </div>
</DocumentFragment>
`;
